### PR TITLE
Openstack integration

### PR DESCRIPTION
**Pull request by prasadadireddi**
Add README doc
Respective dashboard images
Sample conf

### CODE REVIEW

File: collectd-openstack/README.md [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Disk Utilization**: Shows the percentage of utilized disk in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-disk-util.png' width=200px>](./img/chart-openstack-tenant-disk-util.png)
+
+  - **Instances Available vs Used**: Shows the number of available instances and used instances in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-instances.png' width=200px>](./img/chart-openstack-tenant-instances.png)
+
+  - **Memory Usage**: Shows the used memory over available in the tenant/project.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: Not sure what "used memory over available" means. Maybe it's clear if I'm looking at the dashboard, but if this is percentage, then something like "percentage of used memory" could be clearer. Same for the following few entries.

File: collectd-openstack/README.md [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+    [<img src='./img/chart-openstack-tenant-top-memory.png' width=200px>](./img/chart-openstack-tenant-top-memory.png)
+
+  - **Top Instances by VCPUs**: Shows the top five instances by virtual CPUs used in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-top-vcpus.png' width=200px>](./img/chart-openstack-tenant-top-vcpus.png)
+
+
+- **NEUTRON**:
+
+  - **Networks**: Shows the total number of networks created in all prjects.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: typo - projects (fix throughout)

File: collectd-openstack/README.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Routers**: Shows the total number of routers created in all prjects.
+
+    [<img src='./img/chart-openstack-neutron-routers.png' width=200px>](./img/chart-openstack-neutron-routers.png)
+
+  - **Subnets**: Shows the total number of subnets created in all prjects.
+
+    [<img src='./img/chart-openstack-neutron-subnets.png' width=200px>](./img/chart-openstack-neutron-subnets.png)
+
+  - **Flaoting IPs Available vs Used**: Shows the total number of floating IPs used over available in all prjects.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: another instance of "used over available"

File: collectd-openstack/README.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    [<img src='./img/chart-openstack-neutron-floating-ips.png' width=200px>](./img/chart-openstack-neutron-floating-ips.png)
+
+  - **Security Groups Available vs Used**: Shows the total number of security groups used over available in all prjects.
+
+    [<img src='./img/chart-openstack-neutron-security-groups.png' width=200px>](./img/chart-openstack-neutron-security-groups.png)
+
+
+- **INSTANCE**:
+
+  - **VCPUs**: Shows the number of allocated virtual CPUs to the instance.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: number of virtual CPUs allocated to

File: collectd-openstack/README.md [hunk 5]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Total Bytes Sent/Received**: Shows the number of bytes sent and received over the network.
+
+    [<img src='./img/chart-openstack-instance-bytes-sent-received.png' width=200px>](./img/chart-openstack-instance-bytes-sent-received.png)
+
+  - **Received vs Transmitted Packers/sec**: Shows the number of network packets sent and received per second.
+
+    [<img src='./img/chart-openstack-instance-packets-sent-received.png' width=200px>](./img/chart-openstack-instance-packets-sent-received.png)
+
+  - **Virtual Disk Reads and Write Requests**: Shows the number virtual disk read and write request.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: Should this be "number of virtual disk read and write requests"?

File: collectd-openstack/README.md [hunk 6]
```diff
@@ -0,0 +1,300 @@
...
+
+  
+A few other details:
+
+* `plugin` is always set to `openstack`
+* `plugin_instance` will contain the project id and name of the project given in the configuration
+
+
+### METRICS
+Metrics about a hypervisor, tenant and instances are collected by default. For documentation of the metrics and dimensions emitted by this plugin, [click here](./docs). Note, that SignalFx does not support `histogram` and `summary` metric types (hence, metrics of these will be skipped if provided in the configuration). See [usage](#usage) for details.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: ...metrics of these types will be...

File: collectd-openstack/README.md [hunk 7]
```diff
@@ -0,0 +1,300 @@
...
+- [Requirements and Dependencies](#requirements-and-dependencies)
+- [Installation](#installation)
+- [Configuration](#configuration)
+- [Usage](#usage)
+- [Metrics](#metrics)
+- [License](#license)
+
+### DESCRIPTION
+
+This is the SignalFx openstack plugin. Follow these instructions to install the openstack plugin for collectd.
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: updated

File: collectd-openstack/README.md [hunk 8]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Disk Usage**: Shows the used disk over available in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-disk-usage.png' width=200px>](./img/chart-openstack-tenant-disk-usage.png)
+
+  - **Volumes Available vs Used**: Shows the used volumes over available volumes in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-volumes.png' width=200px>](./img/chart-openstack-tenant-volumes.png)
+
+  - **VCPUs Available vs Used**: Shows the used virtaul cpus over available in the tenant/project.
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md [hunk 9]
```diff
@@ -0,0 +1,300 @@
...
+
+| Software  | Version        |
+|-----------|----------------|
+| collectd  |  4.9 or later  |
+| python | 2.6 or later |
+| Python plugin for collectd | (included with [SignalFx collectd agent](https://github.com/signalfx/integrations/tree/master/collectd)[](sfx_link:sfxcollectd)) |
+
+### INSTALLATION
+
+1. Download [collectd-openstack](https://github.com/signalfx/collectd-openstack). Place the `openstack_metrics.py`, `NovaMetrics.py`, `CinderMetrics.py`, and `NeutronMetrics.py` files in `/usr/share/collectd/collectd-openstack`
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: done

File: collectd-openstack/README.md [hunk 10]
```diff
@@ -0,0 +1,300 @@
...
+4. Install the Python requirements with sudo ```pip install -r requirements.txt```
+
+5. Restart collectd
+
+
+### CONFIGURATION
+
+Using the example configuration file [20-openstack.conf](https://github.com/signalfx/integrations/tree/release/collectd-openstack/20-openstack.conf) as a guide, provide values for the configuration options listed below that make sense for your environment and allow you to connect to the openstack instances
+
+| configuration option | definition | example value |
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: done

File: collectd-openstack/README.md [hunk 11]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Physical CPUs**: Shows the number of pysical cores on the host running hypervisor.
+
+    [<img src='./img/chart-openstack-hypervisor-physical-cpus.png' width=200px>](./img/chart-openstack-hypervisor-physical-cpus.png)
+
+  - **Total Disk**: Shows the total disk available on the host running hypervisor.
+
+    [<img src='./img/chart-openstack-hypervisor-total-disk.png' width=200px>](./img/chart-openstack-hypervisor-total-disk.png)
+
+  - **Total Memory**: Shows the total RAM memory available on the host running hypervisor.
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: removed

File: collectd-openstack/README.md [hunk 12]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Used Instances**: Shows the number of used instances in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-used-instances.png' width=200px>](./img/chart-openstack-tenant-used-instances.png)
+
+  - **Memory Utilization**: Shows the percentage of utilized memory in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-memory-util.png' width=200px>](./img/chart-openstack-tenant-memory-util.png)
+
+  - **Disk Utilization**: Shows the percentage of utilized disk in the tenant/project.
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: done

File: collectd-openstack/README.md [hunk 13]
```diff
@@ -0,0 +1,300 @@
...
+
+    [<img src='./img/chart-openstack-tenant-disk-util.png' width=200px>](./img/chart-openstack-tenant-disk-util.png)
+
+  - **Instances Available vs Used**: Shows the number of available instances and used instances in the tenant/project.
```
[the remark this answers is not in the document]
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md (hunk 2 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md (hunk 4 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md (hunk 6 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md (hunk 1 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: Updated the metric description accordingly

File: collectd-openstack/README.md (hunk 3 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: Updated the metric description accordingly

File: collectd-openstack/README.md [hunk 14]
```diff
@@ -0,0 +1,300 @@
...
+* Neutron (Network)
+
+Reference for OpenStack [Monitoring](https://wiki.openstack.org/wiki/Operations/Monitoring).
+
+#### FEATURES
+
+#### Built-in dashboards
+
+
+- **HYPERVISOR**: Provides a high-level overview metrics for an OpenStack hypervisor.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: .. overview of metrics...  (Sorry I missed this the first time)

File: collectd-openstack/README.md [hunk 15]
```diff
@@ -0,0 +1,300 @@
...
+| ---------------------|------------|---------------|
+| ModulePath | Path on disk where collectd can find this module. | "/usr/share/collectd/collectd-openstack/" |
+| AuthURL | Keystone authentication URL/endpoint for the OpenStack cloud | "http://localhost/identity/v3" |
+| Username | Username to authenticate with keystone identity | "admin" |
+| Password | Password to authenticate with keystone identity | "secret" |
+| ProjectName | Specify the name of Project to be monitored | "demo" |
+| ProjectDomainId | Specify the project domain | "default" |
+| UserDomainId | Specify the user domain | "default" |
+| Dimension | Space-separated key-value pair for a user-defined dimension | dimension\_name dimension\_value |
+| Interval | Number of seconds between calls to openstack API. | 10 |
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: to the OpenStack API

File: collectd-openstack/README.md [hunk 16]
```diff
@@ -0,0 +1,300 @@
...
+
+Using the example configuration file [20-openstack.conf](https://github.com/signalfx/integrations/tree/release/collectd-openstack/20-openstack.conf) as a guide, provide values for the configuration options listed below that make sense for your environment and allow you to connect to the openstack instances.
+
+| Configuration option | Definition | Example Value |
+| ---------------------|------------|---------------|
+| ModulePath | Path on disk where collectd can find this module. | "/usr/share/collectd/collectd-openstack/" |
+| AuthURL | Keystone authentication URL/endpoint for the OpenStack cloud | "http://localhost/identity/v3" |
+| Username | Username to authenticate with keystone identity | "admin" |
+| Password | Password to authenticate with keystone identity | "secret" |
+| ProjectName | Specify the name of Project to be monitored | "demo" |
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: name of the project

File: collectd-openstack/README.md [hunk 17]
```diff
@@ -0,0 +1,300 @@
...
+    [<img src='./img/chart-openstack-instance-vda-read-write-size.png' width=200px>](./img/chart-openstack-instance-vda-read-write-size.png)
+
+  
+A few other details:
+
+* `plugin` is always set to `openstack`
+* `plugin_instance` will contain the project id and name of the project given in the configuration
+
+
+### METRICS
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: (this comment applies to line 291) Please check with Akash about the format of this paragraph. I think his script might expect the "For documentation" sentence to be the first, or only, sentence. If you need to change it, the other info could be in a second paragraph.

Comment (prasadadireddi) [reply collectd-openstack/README.md]: Discussed with Akash, and leaving as it is as script would handle it

File: collectd-openstack/README.md (hunk 16 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md (hunk 15 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md (hunk 14 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: corrected

File: collectd-openstack/README.md [hunk 18]
```diff
@@ -0,0 +1,300 @@
...
+|-----------|----------------|
+| collectd  |  4.9 or later  |
+| python | 2.6 or later |
+| Python plugin for collectd | (included with [SignalFx collectd agent](https://github.com/signalfx/integrations/tree/master/collectd)[](sfx_link:sfxcollectd)) |
+
+### INSTALLATION
+
+1. Download [collectd-openstack](https://github.com/signalfx/collectd-openstack). Place the `openstack_metrics.py`, `NovaMetrics.py`, `CinderMetrics.py`, and `NeutronMetrics.py` files into `/usr/share/collectd/collectd-openstack` directory.
+
+2. Modify the [sample configuration file](https://github.com/signalfx/integrations/tree/release/collectd-openstack/20-openstack.conf) for this plugin and copy to `/etc/collectd/managed_config` directory.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: Boy I must need more coffee or something! Each time I review this I see stuff I missed before. Steps 2 and 3 seem to be one step broken into 2 parts. Should it be "Modify the... for this plugin as described in.. below and then copy to...." ?

File: collectd-openstack/README.md [hunk 19]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Disk Usage**: Shows the disk usage free vs used in the hypervisor.
+
+    [<img src='./img/chart-openstack-hypervisor-disk-usage.png' width=200px>](./img/chart-openstack-hypervisor-disk-usage.png)
+
+  - **CPUs and VCPUs**: Shows the number of CPUs available and VCPUs used in the hypervisor.
+
+    [<img src='./img/chart-openstack-hypervisor-cpus-vcpus.png' width=200px>](./img/chart-openstack-hypervisor-cpus-vcpus.png)
+
+  - **Physical CPUs**: Shows the number of pysical cores on the host running hypervisor.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: physical

File: collectd-openstack/README.md [hunk 20]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Disk Usage**: Shows the overall disk space available and used in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-disk-usage.png' width=200px>](./img/chart-openstack-tenant-disk-usage.png)
+
+  - **Volumes Available vs Used**: Shows the maximum number of block storage volumes available and used in the tenant/project.
+
+    [<img src='./img/chart-openstack-tenant-volumes.png' width=200px>](./img/chart-openstack-tenant-volumes.png)
+
+  - **VCPUs Available vs Used**: Shows the maximum number of virtual cpus available and used in the tenant/project.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: CPUs

File: collectd-openstack/README.md [hunk 21]
```diff
@@ -0,0 +1,300 @@
...
+
+  - **Total Bytes Sent/Received**: Shows the number of bytes sent and received over the network.
+
+    [<img src='./img/chart-openstack-instance-bytes-sent-received.png' width=200px>](./img/chart-openstack-instance-bytes-sent-received.png)
+
+  - **Received vs Transmitted Packers/sec**: Shows the number of network packets sent and received per second.
+
+    [<img src='./img/chart-openstack-instance-packets-sent-received.png' width=200px>](./img/chart-openstack-instance-packets-sent-received.png)
+
+  - **Virtual Disk Read and Write Requests**: Shows the number of virtual disk read and write request.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: requests

File: collectd-openstack/README.md (hunk 17 above)
Comment (barbara-sfx) [reply collectd-openstack/README.md]: ok thanks

File: collectd-openstack/README.md [hunk 22]
```diff
@@ -0,0 +1,300 @@
...
+
+  
+A few other details:
+
+* `plugin` is always set to `openstack`
+* `plugin_instance` will contain the project id and name of the project given in the configuration
+
+
+### METRICS
+Metrics about a hypervisor, tenant and instances are collected by default. For documentation of the metrics and dimensions emitted by this plugin, [click here](./docs). Note, that SignalFx does not support `histogram` and `summary` metric types (hence, metrics of these type will be skipped if provided in the configuration). See [usage](#usage) for details.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: of these types

File: collectd-openstack/README.md [hunk 23]
```diff
@@ -0,0 +1,300 @@
...
+
+
+#### Metric naming
+`<metric type>.openstack.<component>.<name of metric>`. This is the format of default metric names reported by the plugin.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: OK as is, but better is "The format of default metric names reported by the plugin is ..."

File: collectd-openstack/README.md [hunk 24]
```diff
@@ -0,0 +1,300 @@
...
+3. Modify the sample configuration file as described in [Configuration](#configuration), below.
+
+4. Install the Python requirements with sudo ```pip install -r requirements.txt```.
+
+5. Restart collectd.
+
+
+### CONFIGURATION
+
+Using the example configuration file [20-openstack.conf](https://github.com/signalfx/integrations/tree/release/collectd-openstack/20-openstack.conf) as a guide, provide values for the configuration options listed below that make sense for your environment and allow you to connect to the openstack instances.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: openstack instances => OpenStack instances

File: collectd-openstack/README.md [hunk 25]
```diff
@@ -0,0 +1,300 @@
...
+* `plugin` is always set to `openstack`
+* `plugin_instance` will contain the project id and name of the project given in the configuration
+
+
+### METRICS
+Metrics about a hypervisor, tenant and instances are collected by default. For documentation of the metrics and dimensions emitted by this plugin, [click here](./docs). Note, that SignalFx does not support `histogram` and `summary` metric types (hence, metrics of these types will be skipped if provided in the configuration). See [usage](#usage) for details.
+
+
+#### Metric naming
+`<metric type>.openstack.<component>.<name of metric>`. The format of default metric names reported by the plugin is.
```
Comment (barbara-sfx) [inline collectd-openstack/README.md]: move the format code to follow the word "is"

Comment (prasadadireddi) [reply collectd-openstack/README.md]: updated

File: collectd-openstack/README.md (hunk 24 above)
Comment (prasadadireddi) [reply collectd-openstack/README.md]: fixed

**Pull request merged**